pkg/apis/servicecatalog/types.go

```diff
@@ -680,63 +680,106 @@ type ServiceInstance struct {
-	// ClusterServicePlanRef below.
+	// as reported by the ClusterServiceBroker. Note that if the
+	// ClusterServiceBroker changes the name of the ClusterServicePlan, it will
+	// not be reflected here, and to see the current name of the
+	// ClusterServicePlan, you should follow the ClusterServicePlanRef below.
 	ClusterServicePlanExternalName string
 
-	// ClusterServiceClassExternalID is the broker's external id for the class.
+	// ClusterServiceClassExternalID is the ClusterServiceBroker's external id
+	// for the class.
 	//
 	// Immutable.
 	ClusterServiceClassExternalID string
 
-	// ClusterServicePlanExternalID is the broker's external id for the plan.
+	// ClusterServicePlanExternalID is the ClusterServiceBroker's external id for
+	// the plan.
 	ClusterServicePlanExternalID string
 
-	// ClusterServiceClassName is the kubernetes name of the
-	// ClusterServiceClass.
+	// ClusterServiceClassName is the kubernetes name of the ClusterServiceClass.
 	//
 	// Immutable.
 	ClusterServiceClassName string
 	// ClusterServicePlanName is kubernetes name of the ClusterServicePlan.
 	ClusterServicePlanName string
+
+	// ServiceClassExternalName is the human-readable name of the
+	// service as reported by the ServiceBroker. Note that if the ServiceBroker
+	// changes the name of the ServiceClass, it will not be reflected here,
+	// and to see the current name of the ServiceClass, you should
+	// follow the ServiceClassRef below.
+	//
+	// Immutable.
+	ServiceClassExternalName string
+	// ServicePlanExternalName is the human-readable name of the plan
+	// as reported by the ServiceBroker. Note that if the ServiceBroker changes
+	// the name of the ServicePlan, it will not be reflected here, and to see
+	// the current name of the ServicePlan, you should follow the
+	// ServicePlanRef below.
+	ServicePlanExternalName string
+
+	// ServiceClassExternalID is the ServiceBroker's external id for the class.
+	//
+	// Immutable.
+	ServiceClassExternalID string
+
+	// ServicePlanExternalID is the ServiceBroker's external id for the plan.
+	ServicePlanExternalID string
+
+	// ServiceClassName is the kubernetes name of the ServiceClass.
+	//
+	// Immutable.
+	ServiceClassName string
+	// ServicePlanName is kubernetes name of the ServicePlan.
+	ServicePlanName string
 }
 
 // ServiceInstanceSpec represents the desired state of an Instance.
 type ServiceInstanceSpec struct {
 	PlanReference
 
 	// ClusterServiceClassRef is a reference to the ClusterServiceClass
-	// that the user selected.
-	// This is set by the controller based on ClusterServiceClassExternalName
+	// that the user selected. This is set by the controller based on
+	// ClusterServiceClassExternalName.
 	ClusterServiceClassRef *ClusterObjectReference
 	// ClusterServicePlanRef is a reference to the ClusterServicePlan
-	// that the user selected.
-	// This is set by the controller based on ClusterServicePlanExternalName
+	// that the user selected. This is set by the controller based on
```

Comment:
Seems like this comment should be updated to include the other ways you can request a `ClusterServicePlan`.